.github/workflows/deploy.yml

```
@@ -7,20 +7,27 @@ on:
 jobs:
   test:
     uses: "ambanum/OpenTermsArchive/.github/workflows/test.yml@main"
+
+  get_instances:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set matrix with instance names
+        id: get_instance_names
+        uses: mikefarah/yq@master
+        with:
+          cmd: yq ".all.children.production.children | keys" ops/inventories/production.yml -o=json | tr '\n' ' '
+      - id: set-matrix
+        run: echo "::set-output name=matrix::{\"instance\":${{ toJson(steps.get_instance_names.outputs.result) }}}"
+
   deploy:
-    needs: [ test ]
+    needs: [ test, get_instances ]
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false  # deploy other instances even if one fails to deploy
-      matrix:
-        instance:
-          - contrib
-          - dating
-          - france
-          - france_elections
-          - pga
-          - p2b_compliance
-
+      matrix: ${{ fromJSON(needs.get_instances.outputs.matrix) }}
```

Do we really need the additional wrapping? Why not:

```suggestion
      matrix:
        instance: ${{ fromJSON(needs.get_instances.outputs.instances) }}

---

I did not manage to make it work like that, and I made many attempts. But maybe I missed something.

Anyway, I'd rather leave it like this now as it works

---

https://github.com/ambanum/OpenTermsArchive/pull/901/commits/a7924981b23937eee0d6dc6febfb476f3a4b690f [seems to work](https://github.com/ambanum/OpenTermsArchive/actions/runs/2875810065) 🙂

---

Well thanks a lot, it's a lot clearer and I though that was one of the first thing I tried (but surely wrongly)